prog/rand.go

```diff
@@ -138,15 +138,33 @@ func (r *randGen) randPageCount() (n uint64) {
 }
 
 // Change a flag value or generate a new one.
-func (r *randGen) flags(vv []uint64, bitmask bool, oldVal uint64) (v uint64) {
+func (r *randGen) flags(vv []uint64, size uint64, bitmask bool, oldVal uint64) (v uint64) {
 	v = oldVal
+	if r.nOutOf(1, 5) {
+		// Ignore the old value sometimes.
+		v = 0
+	}
 	switch {
+	case (bitmask && r.nOutOf(1, 2)):
+		// Try flipping random bits.
+		for stop := false; !stop; stop = r.oneOf(3) {
+			bit := r.rand(int(size) * 8)
+			if r.bin() {
+				v |= 1 << bit
+			} else {
+				v &= ^(1 << bit)
+			}
+		}
 	case (bitmask && r.nOutOf(7, 10)) || (!bitmask && r.nOutOf(1, 5)):
+		// Try combining with randomly chosen flags.
+		// Prioritized when bitmask == true.
 		for stop := false; !stop; stop = r.bin() {
-			v |= vv[r.rand(len(vv))] // prioritized when bitmask = true
+			v |= vv[r.rand(len(vv))]
 		}
 	case (bitmask && r.nOutOf(2, 3)) || (!bitmask && r.nOutOf(7, 8)):
-		v = vv[r.rand(len(vv))] // prioritized when bitmask = false
+		// Chose a random flag.
+		// Prioritized when bitmask == false.
+		v = vv[r.rand(len(vv))]
 	case r.bin():
 		v = 0
```

Comment:
Do we still need this case here since we already have something similar in the beginning?